indico/modules/events/management/program_codes.py

```diff
@@ -49,13 +49,21 @@ def _make_date_placeholder(class_name, name, render_kwarg, date_format, descript
     return type(str(class_name), (DatePlaceholder,), dct)
 
 
-class ContributionIDPlaceholder(Placeholder):
+class ContributionIDPlaceholder(ParametrizedPlaceholder):
     name = 'id'
-    description = _('The ID of the contribution')
+    description = _('The ID of the contribution, optionally padded with zeros to a specified length '
+                    '(e.g. {id:2} for "01" or {id} for "1")')
+    param_required = False
 
     @classmethod
-    def render(cls, contribution):
-        return unicode(contribution.friendly_id)
+    def render(cls, param, contribution):
+        if param is None:
+            return unicode(contribution.friendly_id)
+        try:
+            padding = max(1, min(int(param), 10))
+            return unicode(contribution.friendly_id).zfill(padding)
```

Comment:
I'd move this one into the `else` part of the try..except since errors in here would be exceptional and thus shouldn't be caught